### PR TITLE
Sun/Environment and Font/Outline Color Pickers respect default settings

### PR DESCRIPTION
**Pull request by Guh-Feng**
My attempt for #57154. I did not know how to get clang to work on Visual Studio, but I tried my best to replicate the style.

### CODE REVIEW

**Comment by fire** [comment]
Approved the github actions to test the pull request.

File: editor/plugins/node_3d_editor_plugin.cpp [hunk 1]
```diff
@@ -7370,6 +7370,54 @@ void Node3DEditor::_preview_settings_changed() {
 	}
 }
 
+void Node3DEditor::_picker_settings() {
+	// get default color picker mode from editor settings
```
Comment (KoBeWi) [inline editor/plugins/node_3d_editor_plugin.cpp:7374]:
```suggestion
	// Get default color picker mode from editor settings.
```
Every comment should follow this style.

Although the comments in this PR aren't really necessary, as the code is rather self-explanatory.

File: editor/plugins/node_3d_editor_plugin.cpp [hunk 2]
```diff
@@ -7370,6 +7370,54 @@ void Node3DEditor::_preview_settings_changed() {
...
+		sun_color->get_picker()->set_raw_mode(false);
+
+		if (default_color_mode == 1) {
+			sun_color->get_picker()->set_hsv_mode(true);
+		} else if (default_color_mode == 2) {
+			sun_color->get_picker()->set_raw_mode(true);
+		}
+
+		sun_color->get_picker()->set_picker_shape((ColorPicker::PickerShapeType)picker_shape);
+	}
```
Comment (KoBeWi) [inline editor/plugins/node_3d_editor_plugin.cpp:7390]: Using a sub-scope is not necessary. There aren't any local variables.

**Comment by KoBeWi** [comment]
Overall looks ok, but this much of code makes me think that we should have a single method in EditorNode that initializes a ColorPicker passed as argument and connect it to `about_to_popup` of every ColorPickerButton used in the editor. Could be done in another PR, but it would invalidate the code added here, so idk.

**Comment by Guh-Feng** [comment]
I don't mind taking a shot at that too. I'd like to work on a more elegant solution, if it isn't making it harder to maintain or anything like that.

File: editor/plugins/node_3d_editor_plugin.cpp (hunk 2 above)
Comment (Guh-Feng) [reply editor/plugins/node_3d_editor_plugin.cpp:7390]: It's been a while, so this means that the outer-most braces in portions like this aren't necessary right?

Comment (KoBeWi) [reply editor/plugins/node_3d_editor_plugin.cpp:7390]: Yes, they don't really serve any purpose.

**Comment by KoBeWi** [comment]
What I suggested would actually make it easier to maintain, as you could avoid lots of repeating code. Basically you can extract this part to a separate method:
```C++
picker->set_hsv_mode(false);
picker->set_raw_mode(false);

if (default_color_mode == 1) {
	picker->set_hsv_mode(true);
} else if (default_color_mode == 2) {
	picker->set_raw_mode(true);
}

picker->set_picker_shape((ColorPicker::PickerShapeType)picker_shape);
```
And then do e.g.
```C++
sun_color->get_popup()->connect("about_to_popup", callable_mp(EditorNode::get_singleton(), &EditorNode::setup_popup), varray(sun_color->get_picker()));
```
(not sure about the part with `varray()`)

**Comment by KoBeWi** [comment]
Yeah, you could make a new PR that would supersede this one.

**Comment by fire-forge** [comment]
> Overall looks ok, but this much of code makes me think that we should have a single method in EditorNode that initializes a ColorPicker passed as argument and connect it to about_to_popup of every ColorPickerButton used in the editor. Could be done in another PR, but it would invalidate the code added here, so idk.

Alternatively, there's a proposal for using only one ColorPicker for the entire editor: https://github.com/godotengine/godot-proposals/issues/3857. The settings could be set for this ColorPicker and then they would be respected everywhere because they all use the same one. This would also fix the issue of color presets often being out of sync.

It would be a lot more work to do this than to just fix the issues in these few places, but it would prevent it from happening again in the future. And it would make it a lot easier for plugins to use the default settings as well.

--------------------------

@Guh-Feng I found one other place that doesn't use the default settings in the Theme editor, if you'd like to fix them as well:

image

**Comment by KoBeWi** [comment]
Superseded by #62581

**Pull request closed without merging**